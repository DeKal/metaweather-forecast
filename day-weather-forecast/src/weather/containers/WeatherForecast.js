import { connect } from 'react-redux'
import WeatherForecast from 'weather/pages/WeatherForecast'
import { searchCity } from 'state/city/thunks'
import { getWeatherByCity } from 'state/weather/thunks'

const mapDispatchToProps = (dispatch) => ({
  searchCity: (city) => dispatch(searchCity(city)),
  getWeatherByCity: (cityId) => dispatch(getWeatherByCity(cityId))
})

export default connect(null, mapDispatchToProps)(WeatherForecast)
