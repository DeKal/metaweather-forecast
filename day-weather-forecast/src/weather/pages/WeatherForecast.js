import React from 'react'
import PropTypes from 'prop-types'
import { Grid, Container, makeStyles } from '@material-ui/core'
import debounce from 'lodash/debounce'
import WeatherPanel from 'weather/containers/WeatherPanel'
import SearchBar from 'weather/containers/SearchBar'

const useStyles = makeStyles((theme) => ({
  search: {
    marginTop: theme.spacing(4)
  }
}))

const WeatherForecast = ({ searchCity, getWeatherByCity }) => {
  const classes = useStyles()
  return (
    <Container maxWidth="lg">
      <Grid container spacing={4}>
        <Grid item sm={6} xs={12}>
          <SearchBar
            placeholder="London"
            className={classes.search}
            fullWidth
            onChange={(city) => {
              if (city) {
                getWeatherByCity(city.woeid)
              }
            }}
            onChangeInput={debounce(function makeQuery(query) {
              searchCity(query)
            }, 200)}
          />
        </Grid>
        <Grid item xs={12}>
          <WeatherPanel />
        </Grid>
      </Grid>
    </Container>
  )
}
export default WeatherForecast
WeatherForecast.propTypes = {
  searchCity: PropTypes.func.isRequired,
  getWeatherByCity: PropTypes.func.isRequired
}
