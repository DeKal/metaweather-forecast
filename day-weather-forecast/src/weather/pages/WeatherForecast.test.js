import React from 'react'
import { fireEvent, getByTestId, render } from '@testing-library/react'
import WeatherForecast from 'weather/pages/WeatherForecast'

jest.mock('weather/containers/WeatherPanel')
jest.mock('weather/containers/SearchBar')

describe('WeatherForecast test', () => {
  test('should render correctly', () => {
    const getWeatherByCityMock = jest.fn()
    const { container } = render(
      <WeatherForecast
        searchCity={() => {}}
        getWeatherByCity={getWeatherByCityMock}
      />
    )
    expect(container).toMatchSnapshot()

    const input = getByTestId(container, 'mock-search-bar')
    fireEvent.change(input, {
      target: { value: 111 }
    })

    expect(getWeatherByCityMock).toBeCalledWith('111')
  })
})
