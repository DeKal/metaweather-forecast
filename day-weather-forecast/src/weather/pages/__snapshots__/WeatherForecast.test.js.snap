// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForecast test should render correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <input
          data-testid="mock-search-bar"
          type="text"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div>
           Mock WeatherPanel
        </div>
      </div>
    </div>
  </div>
</div>
`;
